server/interpreter.js

```diff
@@ -1467,25 +1597,27 @@ Interpreter.prototype.createFunction = function(node, scope) {
 /**
  * Create a new native function.
  * @param {!Function} nativeFunc JavaScript function.
- * @param {boolean=} opt_constructor If true, the function's
- * prototype will have its constructor property set to the function.
- * If false, the function cannot be called as a constructor (e.g. escape).
- * Defaults to undefined.
+ * @param {!Interpreter.Object|boolean=} prototype If an object is
+ *     supplied, that object will be added as the function's
+ *     .prototype property (with the object receivieng a corresponding
+ *     .constructor properyt).  If true, a new ordinary
+ *     Interpreter.Object will be created for the purpose.  If false
```

Comment:
Done.  I also realised that there were some related problems with the code in `initRegExp` and `initDate` that called this function with `prototype === true`, which are also now fixed.